test/unit/network_utils_spec.js

```diff
@@ -238,6 +238,13 @@ describe('network_utils', function() {
         }
         throw new Error(`Unexpected headerName: ${headerName}`);
       })).toEqual('filename.pdf');
+
+      expect(extractFilenameFromHeader((headerName) => {
```

Comment:
I think it would also be good to add another test case for a filename with a % character in it, but that's not actually URL encoded, for example `100%.pdf`. In that case, no transformation should be done.

Reply:
test cases have been added.